### ISSUE TITLE
long method and variable declarations not correctly indented

### ISSUE THREAD

**Issue by gmaurel** [uncrustify/uncrustify#672]
hi Ben,

The fix for 536 solves the problem for the class declarations. 

Long method and variable declarations though are still not being indented correctly. They work normally, but in case of templatized parameters they break. Adding an additional check should handle this problem.
Around line 1430 in indent.cpp we have this condition check. We need to add this additional check. See comment alongside the new condition.

<code>
     else if ((vardefcol > 0) &&
              (pc->level == pc->brace_level) &&
              ((pc->type == CT_WORD) || (pc->type == CT_FUNC_CTOR_VAR)) &&
              (prev != NULL) &&
              ((prev->type == CT_COMMA) ||
               (prev->type == CT_TYPE) ||
(cpd.lang_flags & LANG_JAVA && prev->parent_type == CT_TEMPLATE) || // New condition added
(prev->type == CT_WORD)) &&
((pc->flags & PCF_VAR_DEF) != 0))
{
LOG_FMT(LINDENT, "%s: %d] Vardefcol => %d\n",
func, pc->orig_line, vardefcol);
reindent_line(pc, vardefcol);
}
</code>

For the long method problem though, I am afraid I don't have a solution.

Reported by: kaiser101

Original Ticket: [uncrustify/bugs/555](https://sourceforge.net/p/uncrustify/bugs/555)

**Comment by kaiser101** [comment]
Can someone help me with this? Everything seems to be working except class definition statements. Long method declarations, statements all work fine with indent_continue.

Checking logs with an earlier version of uncrsutify suggests that there are 2 passes. First is in indent.cpp when the line is split and indent continue comes into play. In this case the next line is indented correctly i.e. using the value set in indent_continue. But then there is a second pass in one of the cleanup.cpp files, brace or tokenize I dont remember. Thats where the indent for the split line is set back to the original indent, i.e. the indent_continue value is disregarded.

In a nutshell the feature is expected to behave as follows for class declarations
indent_columns = 4
indent_continue = 8
code_width = 80

Input
public abstract class KeyValueItemWriter<K, V> implements ItemWriter<V>, InitializingBean {

Expected output
public abstract class KeyValueItemWriter<K, V> implements ItemWriter<V>,
        InitializingBean { // indented by 8 spaces, instead of 4

Actual output // incorrect
public abstract class KeyValueItemWriter<K, V> implements ItemWriter<V>,
    InitializingBean { // indented by 4 spaces

**Comment by guy-maurel** [comment]
Duplicate to #1109

**Comment by guy-maurel** [comment]
I am working on. It is not only with java sources.

**Comment by kaiser101** [comment]
Thank you so much Ben. Really appreciate it. It is the only thing holding us back from using it.

**Comment by kaiser101** [comment]
Hi Guy, Give me a day or two. I will test it and revert. Many thanks.

**Comment by guy-maurel** [comment]
Please use the new option UO_indent_continue_class_head

**Comment by kaiser101** [comment]
Hi guy, I am on a windows box and I am using msys to do the build, but I am getting the following error when I try to build.

/d/guymaurel/uncrustify/src/punctuators.cpp: In function ‘const chunk_tag_t* find_punctuator(const char*, int)’:
/d/guymaurel/uncrustify/src/punctuators.cpp:48:32: error: ‘punc_table’ was not declared in this scope
    const auto        *parent = punc_table; //!< graph in table form, initially point at first entry
                                ^~~~~~~~~~
/d/guymaurel/uncrustify/src/punctuators.cpp:48:32: note: suggested alternative: ‘mutable’
    const auto        *parent = punc_table; //!< graph in table form, initially point at first entry
                                ^~~~~~~~~~
                                mutable
make[2]: *** [CMakeFiles/uncrustify.dir/build.make:743: CMakeFiles/uncrustify.dir/src/punctuators.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:932: CMakeFiles/uncrustify.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

**Comment by kaiser101** [comment]
Hi Guy, I managed to resolve the issue with the build. It works great except that I think there is a regression somewhere or I have my config screwed up.

```
public class SavingsInvestingParticipantDataFileItemReader implements
		Runnable<FileItemReader> {
    public static void main<FileItemReader>(String[] args, FileItemReader
		    reader ) {
    for (int i = 0; i < 20; i++) {
	System.out.println( "This is a test." );
    }
    }
}
```
Have a look at the for within the method. Shouldn't it be indented by 4 spaces?

**Comment by kaiser101** [comment]
[uncrustify.txt](https://github.com/uncrustify/uncrustify/files/2063146/uncrustify.txt)

**Comment by kaiser101** [comment]
Have attached my config for reference. Could you please have a look?

**Comment by guy-maurel** [comment]
yes, soon.

**Comment by guy-maurel** [comment]
This is correct.
You are using: _indent_class  = true_
The function _public static void main..._ is a member of the class and will be intended

**Comment by kaiser101** [comment]
Hi Guy, I was referring to the for loop within the main method. It should be indented further by 4 spaces, but it is not.

**Comment by kaiser101** [comment]
Ok got it, thanks working now.

**Comment by gmaurel** [comment]
The bug is fixed #1765
